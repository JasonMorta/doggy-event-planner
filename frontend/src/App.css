.App {
  min-height: 100vh;
  display: flex;
  width: 42.125rem; 
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  /* font-size: calc(10px + 2vmin);
  color: white; */
  position: relative;
  color: gray;
  margin: auto;
}

.App a {
  color: #61dafb;
}




.MuiDialog-paper {
  /* background-image: url('./static/images/logInModal.png');
  background-size: 103%;
  width: 451px;
  border-radius: 35px !important;
  color: black !important;
  background-position-y: center;
  background-repeat: no-repeat; */
  background-color: #f5ffdb !important;
}


html {
  width: max-content;
      margin: auto;
    background-color: #f2f2f2;
}