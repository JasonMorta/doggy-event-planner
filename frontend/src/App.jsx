
import './App.css';
import { createContext } from 'react';
import { useState } from 'react';
import Dash from './pages/Dash';
import Shape from './components/spinner/shadpe-devider/Shape';
import NavBar from './components/navigation/NavBar';
import { BrowserRouter as Router, Route, Routes} from "react-router-dom";
import About from './pages/About';
import ErrorPage from './pages/ErrorPage';
import Gallery from './pages/Gallery';



//create context hook
//This hook allow any nested children to share and alter data with the use of props.
export const sharedState = createContext()

function App() {

  //create a state for every value that needs to be changed
  //these values can be changed from the parent or any child component
  const [currentUser, setCurrentUser] = useState('');
  const [userRoll, setUserRoll] = useState('visitor');
  const [loggedIn, setLoggedIn] = useState(false);
  const [dogOwners, setDogOwners] = useState([]);
  const [allEvents, setAllEvents] = useState([]);

  
  const [editButton, setEditButton] = useState();
  const [eventId, setEventId] = useState("")
  //Store selected event id in state
  const [thisEvent, setThisEvent] = useState({});
  const [update, setUpdate] = useState(false);
  const [thisComment, setThisComment] = useState("")
  const [comments, setComments] = useState([]);
  const [commentId, setCommentId] = useState('');


  return (
    //To pass and sharedState with all components, we use the .Provider element and pass data through value={[stateValues]}
    <sharedState.Provider value={[loggedIn, setLoggedIn, dogOwners, setDogOwners, allEvents, setAllEvents, comments, setComments, editButton, setEditButton, eventId, setEventId, thisEvent, setThisEvent,update, setUpdate, thisComment, setThisComment, commentId, setCommentId, currentUser, setCurrentUser,userRoll, setUserRoll ]}>
    <div className="App">
      
    


      <Router>
      <NavBar />
        <Routes>
          <Route path="/" element={<Dash />} />
          <Route path="/About" element={<About />} />
          <Route path="/Gallery" element={<Gallery />} />
          <Route path="*" element={<ErrorPage />} />
        </Routes>
     
      </Router>

    </div>
    </sharedState.Provider>
  );
}

export default App;
