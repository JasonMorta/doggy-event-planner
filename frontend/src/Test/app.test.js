import React from 'react';
import App from '../App.jsx'

import renderer from 'react-test-renderer';

// use 'npm test' for frontend testing
it('App renders correctly', () => {
   const tree = renderer
   .create(<App/>)
   expect(tree).toMatchSnapshot();
   });