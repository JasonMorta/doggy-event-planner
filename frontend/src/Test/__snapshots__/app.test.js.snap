// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="nav-container"
  >
    <div
      className="navBar MuiBox-root css-0"
    >
      <div
        className="MuiBottomNavigation-root css-q7d5b0-MuiBottomNavigation-root"
      >
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-10dswpr-MuiButtonBase-root-MuiBottomNavigationAction-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="HomeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          <span
            className="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
          >
            Home
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root css-10dswpr-MuiButtonBase-root-MuiBottomNavigationAction-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PanoramaIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23 18V6c0-1.1-.9-2-2-2H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2zM8.5 12.5l2.5 3.01L14.5 11l4.5 6H5l3.5-4.5z"
            />
          </svg>
          <span
            className="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Gallery
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="dashboard-container"
  >
    <div
      className="signUp-container"
    >
      <div
        className="logIn-btn"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium com-btn css-17n64wm-MuiButtonBase-root-MuiButton-root"
          data-testid="reusable-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Join
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium com-btn css-170d7yb-MuiButtonBase-root-MuiButton-root"
          data-testid="reusable-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          LOG IN
        </button>
      </div>
    </div>
    <div
      className="dashboard"
    >
      <div
        className="event-main-container"
      >
        <h1>
          Weekly Events
        </h1>
        <div
          className="skeletons"
        >
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            No events at the moment
          </h3>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            check back later
          </h3>
          <div
            className="placeholder"
          >
            <div
              className="MuiBox-root css-1v5z18m"
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <br />
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
                style={
                  Object {
                    "height": 118,
                    "width": 300,
                  }
                }
              />
            </div>
          </div>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            No events at the moment
          </h3>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            check back later
          </h3>
          <div
            className="placeholder"
          >
            <div
              className="MuiBox-root css-1v5z18m"
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <br />
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
                style={
                  Object {
                    "height": 118,
                    "width": 300,
                  }
                }
              />
            </div>
          </div>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            No events at the moment
          </h3>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            check back later
          </h3>
          <div
            className="placeholder"
          >
            <div
              className="MuiBox-root css-1v5z18m"
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <br />
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
                style={
                  Object {
                    "height": 118,
                    "width": 300,
                  }
                }
              />
            </div>
          </div>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            No events at the moment
          </h3>
          <h3
            style={
              Object {
                "margin": 0,
                "textAlign": "center",
              }
            }
          >
            check back later
          </h3>
          <div
            className="placeholder"
          >
            <div
              className="MuiBox-root css-1v5z18m"
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <br />
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
                style={
                  Object {
                    "height": 118,
                    "width": 300,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="commentBlock"
      >
        <img
          alt="dag waving"
          className="hi-dog"
          src="hi.png"
        />
        <h3>
          Add Your Comments
        </h3>
        <p>
          You must be logged in to add comments
        </p>
        <ul
          className="MuiList-root MuiList-padding commentsList css-meneo4-MuiList-root"
        />
      </div>
    </div>
    <div
      className="about-container"
    >
      <div
        className="headingText"
      >
        <h1
          className="jello-vertical"
          style={
            Object {
              "animationDuration": "3.0s",
              "color": "#7f8aff",
            }
          }
        >
          W
        </h1>
        <h1
          className="jello-vertical"
          style={
            Object {
              "animationDuration": "3.1s",
              "color": "#ff7ca3",
            }
          }
        >
          H
        </h1>
        <h1
          className="jello-vertical"
          style={
            Object {
              "animationDuration": "3.2s",
              "color": "#7f8aff",
            }
          }
        >
          Y
        </h1>
      </div>
      <p>
        Some dogs need socialization, others may need a calm fiend, or even just one that's the same size. Whether they’re tiny, wild, anxious or old, they deserve a time and space where they can feel secure and free with other dogs of a similar size and/or energy level.
      </p>
      <p>
        A safe space for your furbabies to plan meetups, Socialize, and Play.
      </p>
      <p>
        Different groups for different needs (mini, active, giant, anxious,...)
      </p>
      <p>
        Save time by knowing exactly when meet-ups are happening.
      </p>
      <p>
        Connect with other dog parents.
      </p>
    </div>
  </div>
</div>
`;
